### ISSUE TITLE
Challenging Cases

### ISSUE THREAD

**Issue by wingman-jr-addon** [wingman-jr-addon/model#4]
Note some of these contain objectionable content:

- https://github.com/wangx404/anti-NSFW-detection-test/tree/master/images

Translations for relevant portions of failing cases even on "untrusted" level from `anti-NSFW-detection`:

> 2. Stitching blank images
> 
> In most cases, CNN has limitations on the size of the input picture, so when detecting the picture, it will generally be scaled to a 1:1 square picture first, and then the detection is performed. For such a picture preprocessing method, we can splice the original picture and a longer blank picture together to form a picture with an aspect ratio far greater than 1. Such a picture will deviate from the data distribution of the model training data set after being scaled to a square, so it can play a role in confrontation.
> 
> However, this approach can also be quickly repaired. For pictures whose aspect ratio deviates greatly from 1:1, we only need to divide the picture into multiple 1:1 pictures and then detect them separately.
> 
> For such a confrontation method, it can be seen from the diagram below that Baidu AI and Tencent AI may have completely different detection results due to different processing methods. 
> 
> ...
> 
> P.S. At present, Sina's algorithm has not made targeted improvements to this confrontation method. P.P.S. Splicing multiple pictures together can also counter the detection of algorithms, but since most platforms will greatly compress long pictures, there will be more serious quality loss, so it may be better to splice blank pictures. P.P.P.S. Such a confrontational idea should not be difficult to think of, but because the targeted processing method will increase the amount of calculation several times, this may be the reason why the platform has not been improved. 
> 
> 
> 3. Convert to gif image
> 
> A blank picture and a target picture are respectively used as two frames of the gif picture, the blank frame is set to a shorter duration, and the target picture is set to a longer duration. The effect of the gif image obtained in this way when clicked to view is very small from the normal image (the disadvantage is that the volume of the image is greatly increased).
> 
> Such a confrontation method is very easy to detect in theory. The algorithm only needs to analyze the gif frame by frame, analyze and calculate it frame by frame, and finally return the maximum probability.
> 
> However, the fact is that Baidu AI provides detection for gif pornographic images (the interface description: this request is used to identify the pornographic degree of GIF images. For non-gif interfaces, please use the image review combination interface. The interface will check every frame in the picture. Recognize and return the result with the highest pornographic value among all the detection results. Currently, three dimensions are supported: pornography, sexy, and normal.) Tencent AI does not support gif format pictures, and Sina Weibo has not improved this.
> 
> 4. Stripe & gif
> 
> According to a certain stripe interval, the horizontal pixel value of the picture is alternately set to 255/0.
> 
> Unless similar image augmentation methods are added in the training process, it is difficult for the algorithm to accurately determine the category of such images. But the problem with such countermeasures is that the look and feel of the picture will be affected to a certain extent. The wider the stripe width, the more difficult it is to be detected, but the worse the look and feel caused by the loss of information will be more serious. In order to alleviate this problem, it may be necessary to superimpose two striped pictures together to form a gif to make up for this defect.
> 
> When the stripe width is 1, the detection result of Tencent AI is normal; when the stripe width is 16, the detection result of Baidu AI is also normal.
> After splicing two striped picture frames together to form a gif, although the look and feel is slightly improved, the stroboscopic flicker caused by the limitation of the gif frame rate will cause discomfort

**Comment by wingman-jr-addon** [comment]
The animation cases have been addressed by adding true GIF support to the addon (which is distinct from the issue of the model here).
